plugins/serializers/customjson/customjson.go

```diff
@@ -0,0 +1,168 @@
+package customjson
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/influxdata/telegraf"
+	"github.com/jmespath/go-jmespath"
+)
+
+type serializer struct {
+	JmespathExpression string
+	TagsPrefix         string
+}
+
+func NewSerializer(jmespath_expression string, tags_prefix string) (*serializer, error) {
+	s := &serializer{
+		JmespathExpression: jmespath_expression,
+		TagsPrefix:         tags_prefix,
+	}
+	return s, nil
+}
+
+func (s *serializer) Serialize(metric telegraf.Metric) ([]byte, error) {
+
+	m, err := s.createObject(metric)
+	if err != nil {
+		return nil, fmt.Errorf("D! [serializer.customjson] Dropping invalid metric: %s", metric.Name())
+	}
+
+	return m, nil
+}
+
+func (s *serializer) SerializeBatch(metrics []telegraf.Metric) ([]byte, error) {
+
+	var serialized []byte
+
+	for _, metric := range metrics {
+		m, err := s.createObject(metric)
+		if err != nil {
+			return nil, fmt.Errorf("D! [serializer.customjson] Dropping invalid metric: %s", metric.Name())
+		} else if m != nil {
+			serialized = append(serialized, m...)
+		}
+	}
+
+	return serialized, nil
+}
+
+func (s *serializer) createObject(metric telegraf.Metric) (metricGroup []byte, err error) {
+
+	/*  All fields index become dimensions and all tags index can be prefixed by tags_prefix config input and located on json root.
+	    ** Default customjson format contains the following fields:
+		** metric_family: The name of the metric
+		** metric_name:   The name of the fields dimension
+		** metric_value:  The value of the fields dimension
+		** *:             The name and values of the tags
+		** timestamp:     The timestamp for the metric
+	*/
+
+	// Build output result
+	dataGroup := map[string]interface{}{}
+	var metricJson []byte
+
+	if s.JmespathExpression != "" {
+		jmespath.MustCompile(s.JmespathExpression)
+	}
+
+	for _, field := range metric.FieldList() {
+
+		fieldValue, valid := verifyValue(field.Value)
+
+		if !valid {
+			log.Printf("D! Can not parse value: %v for key: %v", field.Value, field.Key)
+			continue
+		}
+
+		// Build root parameter
+		dataGroup["metric_family"] = metric.Name()
+		// Convert ns to float milliseconds since epoch.
+		dataGroup["timestamp"] = float64(metric.Time().UnixNano()) / float64(1000000)
+
+		// Build fields parameter
+		dataGroup["metric_name"] = field.Key
+		dataGroup["metric_value"] = fieldValue
+
+		// Build tags parameter
+		if s.JmespathExpression != "" {
+			metricJson, err = buildJmespathTagsParameter(metric, dataGroup, s)
+		} else {
+			metricJson, err = buildDefaultTagsParameter(metric, dataGroup, s)
+		}
```

Comment:
You might want to do this decision in an `Init()` function and store the function pointer in the serializer struct...